refactor(kitchenowl): resolve issue #8804

polarstack opened the pull request:
**Description**
Kitchenowl was migrated to new common, but deploying ended in "Hmmmm... couldn't reach server" error and "(111: Connection refused) while connecting to upstream" log entries. 
This PR removes the deprecated "additionalContainer" section and deploys the backend as a Deployment. Furthermore the secrets template was refactored to render it instead of hardcoding. Additionally cnpg was added as it is supported by the upstream project.
As the current chart is not deployable and due to the refactoring of several parts I was not able to test, but I expect it to be a breaking change for old deployments as the data is now stored in cnpg instead of sqlite.

⚒️ Fixes  #8804

**⚙️ Type of change**

- [ ] ⚙️ Feature/App addition
- [x] 🪛 Bugfix
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
Tested on TrueNAS Scale 22.12.3.2 with all operators and best practises according getting started docs (including Ingress).
After deployment all pods including cnpg are running, no errors in the logs and application events

**📃 Notes:**

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [x] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

polarstack committed:
fix(kitchenowl) make it deployable again

Remove old additionalContainers, add probes, add cnpg

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
refactor(kitchenowl) secrets tpl

refactore secrets template

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
refactor(kitchenowl) common template

adjust kitchenowl common template to secrets template

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
update(kitchenowl) version to new major

Bump Major Release after refactoring chart

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack commented:
Did the re-factoring according to feedback in https://github.com/truecharts/charts/pull/11191
Sorry for opening a new PR instead of adjusting the old one - my bad... 
=> Now I know Contribution Guideline: "Don't close when a reviewer requests changes (just push the changes or ask for help)."

polarstack committed:
Merge branch 'truecharts:master' into master

polarstack committed:
Merge branch 'truecharts:master' into master

polarstack committed:
Merge branch 'master' into master

StevenMcElligott commented on charts/stable/kitchenowl/values.yaml:
```diff
@@ -6,51 +6,119 @@ backendImage:
-            # Backend listens on 5000 websockets.
-            BACK_URL: "localhost:5000"
+            FRONT_URL: "{{ .Values.workload.main.podSpec.containers.main.env.FRONT_URL }}"
+            BACK_URL: "kitchenowl-backend:{{ .Values.service.backend.ports.backend.port }}"
```
I feel like this is off, maybe just me, it's not connecting to nginx

polarstack replied:
You were right, the first part of BACK_URL was hardcoded to "kitchenowl-backend" as what it is called when it gets installed directly on TrueNAS. But in the Github checks the Pod gets a slightly different name: kitchenowl-randomstring-backend-57485bcd8-2mq7j. I fixed that with moving BACK_URL to a configmap and generate it with the $fullname variable. Finally all checks are successful :-)

StevenMcElligott replied:
Huge!

polarstack committed:
refactor(kitchenowl) add configmap for back_url

refactor BACK_URL to get value from a configmap

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
add(kitchenowl) configmap template

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
add(kitchenowl) configmap render

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
Merge branch 'truecharts:master' into master

polarstack committed:
fix(kitchenowl) configmap linting

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
add(kitchenowl) move back_url to envFrom

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
fix(kitchenowl) configMapRef instead of configMapKeyRef

When I added the configmap first I put a configMapKeyRef in the env section. Forgot to change it to configMapRef

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
fix(kitchenowl) typos in common.yaml

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
fix(kitchenowl) typos in configmap.tpl

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
fix(kitchenowl) order of envFrom

Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
Merge branch 'master' into master

Crow-Control reviewed (approved):
This is generally some damn nice work!
Any why was this merged into stable earlier >.<

truecharts-admin commented:
This PR is locked to prevent necro-posting on closed PRs. Please create a issue or contact staff on discord if you want to further discuss this
